packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -417,12 +417,18 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             }
         } while (selected && exist && !overwrite);
         if (selected) {
+            const temp = new URI(uri.toString() + '(copy).tmp');
             try {
+                await this.fileService.copy(uri, temp);
                 await this.commandRegistry.executeCommand(CommonCommands.SAVE.id);
-                await this.fileService.copy(uri, selected, { overwrite });
+                await this.fileService.move(uri, selected, { overwrite });
+                overwrite = true;
+                await this.fileService.move(temp, uri, { overwrite });
             } catch (e) {
                 console.warn(e);
             }
+            await this.commandRegistry.executeCommand(CommonCommands.CLOSE_TAB.id);
+            await open(this.openerService, selected);
```

Comment:
Do we want to perform these operations even if there is an error in the `try-catch`?
Would it not be more valid to only perform these operations if the `save` was successful?

Reply:
Yes, it makes more sense that way. I put the statements inside the try block.